### PR TITLE
feat : add merkle tree wrapper

### LINKED ISSUE #90: [merkle tree] make new merkle tree wrapper
**Proposal Detail**
the implementation of merkle tree `https://github.com/NebulousLabs/merkletree` affect original codes. 
I think `https://github.com/cbergoon/merkletree` doesn't have much affect out code. 

**Reference**
https://github.com/cbergoon/merkletree

### PR DESCRIPTION
**Pull request by AgwaB**
## Related Issue

resolve: #90 

## Description

- add merkle tree wrapper
- modify proto buffer field : move epoch to message
## Checklist

Thank you for your contribution.
Before submitting this PR, please make sure:

- [x] Applying goimports
- [x] Test case
- [x] End of Work

### CODE REVIEW

File: rbc/merkle_tree.go [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+	for _, data := range datas {
+		merkleDatas = append(merkleDatas, Data(data))
+	}
+
+	tree, err := merkletree.NewTree(merkleDatas)
+	if err != nil {
+		return nil, err
+	}
+
+	return tree, nil
```
Comment (zeroFruit) [inline rbc/merkle_tree.go]: plz wrapping this tree, so that we do not expose `merkletree.MerkleTree` to exteranl

**Comment by GunHee5719** [comment]
LGTM:)

File: rbc/merkletree.go [hunk 2]
```diff
@@ -0,0 +1,8 @@
+package rbc
+
+type RootPath [][]byte
+
+type Merkletree interface {
```
Comment (zeroFruit) [inline rbc/merkletree.go]: plz remove this

File: rbc/merkletree/merkletree.go [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+
+func (w *Wrapper) MerkleRoot() []byte {
+	return w.MerkleRoot()
+}
+
+func (w *Wrapper) MerklePath(data []byte) (RootPath, []int64, error) {
+	return w.tree.GetMerklePath(Data(data))
+}
+
+type Data []byte
```
Comment (zeroFruit) [inline rbc/merkletree/merkletree.go]: should change it to private with more meaningful name

File: rbc/merkletree/merkletree.go [hunk 4]
```diff
@@ -0,0 +1,124 @@
...
+	}
+
+	return h.Sum(nil), nil
+}
+
+func (c content) Equals(other merkletree.Content) (bool, error) {
+	return bytes.Equal(c, other.(content)), nil
+}
+
+func New(datas []Data) (*Wrapper, error) {
```
Comment (zeroFruit) [inline rbc/merkletree/merkletree.go]: New(dataList []Data)

File: rbc/merkletree/merkletree.go [hunk 5]
```diff
@@ -0,0 +1,124 @@
...
+
+	t, err := merkletree.NewTree(merkleDatas)
+	if err != nil {
+		return nil, err
+	}
+
+	return &Wrapper{*t}, nil
+}
+
+func ValidatePath(data Data, rootHash []byte, rootPath RootPath, indexes []int64) bool {
```
Comment (zeroFruit) [inline rbc/merkletree/merkletree.go]: indexes -> indexList

File: rbc/merkletree/merkletree.go [hunk 6]
```diff
@@ -0,0 +1,124 @@
...
+
+	return h.Sum(nil), nil
+}
+
+func (c content) Equals(other merkletree.Content) (bool, error) {
+	return bytes.Equal(c, other.(content)), nil
+}
+
+func New(datas []Data) (*Wrapper, error) {
+	var merkleDatas []MerkleData
```
Comment (zeroFruit) [inline rbc/merkletree/merkletree.go]: merkleDataList

**Review by zeroFruit — APPROVED** [review]
Nice work!

**Pull request merged**